PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h

```diff
@@ -271,3 +272,134 @@ class FirstObjectSimpleFlatTableProducer : public SimpleFlatTableProducerBase<T,
+        skipNonExistingSrc_(
+            params.existsAs<bool>("skipNonExistingSrc") ? params.getParameter<bool>("skipNonExistingSrc") : false),
+        src_(skipNonExistingSrc_ ? mayConsume<TProd>(params.getParameter<edm::InputTag>("src"))
+                                 : consumes<TProd,edm::InLumi>(params.getParameter<edm::InputTag>("src"))) {
+    edm::ParameterSet const &varsPSet = params.getParameter<edm::ParameterSet>("variables");
+    for (const std::string &vname : varsPSet.getParameterNamesForType<edm::ParameterSet>()) {
+      const auto &varPSet = varsPSet.getParameter<edm::ParameterSet>(vname);
+      const std::string &type = varPSet.getParameter<std::string>("type");
+      if (type == "int")
+        vars_.push_back(std::make_unique<IntVar>(vname, varPSet));
+      else if (type == "float")
+        vars_.push_back(std::make_unique<FloatVar>(vname, varPSet));
+      else if (type == "uint8")
+        vars_.push_back(std::make_unique<UInt8Var>(vname, varPSet));
+      else if (type == "bool")
+        vars_.push_back(std::make_unique<BoolVar>(vname, varPSet));
+      else
+        throw cms::Exception("Configuration", "unsupported type " + type + " for variable " + vname);
+    }
+
+    produces<nanoaod::FlatTable,edm::Transition::EndLuminosityBlock>();
+  }
+
+  ~SimpleFlatTableProducerBaseLumi() override {}
+
+  // this is to be overriden by the child class
+  virtual std::unique_ptr<nanoaod::FlatTable> fillTable(const edm::LuminosityBlock &iLumi,
+                                                        const edm::Handle<TProd> &prod) const = 0;
+
+  void produce(edm::Event &iEvent, const edm::EventSetup &iSetup) override {
+    // do nothing 
+  }
+
+  void endLuminosityBlockProduce(edm::LuminosityBlock& iLumi, const edm::EventSetup &iSetup) final {
+    edm::Handle<TProd> src;
+    iLumi.getByToken(src_, src);
+
+    std::unique_ptr<nanoaod::FlatTable> out = fillTable(iLumi, src);
+    out->setDoc(doc_);
+
+    iLumi.put(std::move(out));
+  }
+
+protected:
+  const std::string name_;
+  const std::string doc_;
+  const bool extension_;
+  const bool skipNonExistingSrc_;
+  const edm::EDGetTokenT<TProd> src_;
+
+  class VariableBase {
+  public:
+    VariableBase(const std::string &aname, const edm::ParameterSet &cfg)
+        : name_(aname),
+          doc_(cfg.getParameter<std::string>("doc")),
+          precision_(cfg.existsAs<int>("precision") ? cfg.getParameter<int>("precision")
+                                                    : (cfg.existsAs<std::string>("precision") ? -2 : -1)) {}
+    virtual ~VariableBase() {}
+    const std::string &name() const { return name_; }
+
+  protected:
+    std::string name_, doc_;
+    int precision_;
+  };
+
+  class Variable : public VariableBase {
+  public:
+    Variable(const std::string &aname, const edm::ParameterSet &cfg) : VariableBase(aname, cfg) {}
+    virtual void fill(std::vector<const T *> selobjs, nanoaod::FlatTable &out) const = 0;
```

Comment:
Can the copy of the vector be avoided?
```suggestion
    virtual void fill(const std::vector<const T *>& selobjs, nanoaod::FlatTable &out) const = 0;
```

Reply:
Dear @makortel,

I based the 'Lumi version' of the module off the existing _SimpleFlatTableProducerBase_: https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h#L85

Therefore, I think this question is best addressed to the original developers. If it were to be updated, then I think it would make sense that it is updated consistently in both modules by the original developers.

Best,
Mateusz